TwoPageEBookReader/app/src/main/java/com/example/ebook_reader_sample/BookActivity.kt

```diff
@@ -0,0 +1,258 @@
+        bookPagerView.setCurrentItem(position, false)
+        bookPagerView.registerOnPageChangeCallback(pagePagerCallback)
+        bookPagerView.isUserInputEnabled = true
+    }
+
+    inner class BookPagerCallback() : ViewPager2.OnPageChangeCallback() {
+        override fun onPageSelected(position: Int) {
+            when (position) {
+                0 -> {
+                    if (book.currentChapter > 0) {
+                        bookPagerView.isUserInputEnabled = false
+                        handler.postDelayed({
+                            book.currentChapter = book.currentChapter - 1
+                            book.currentPage = book.numPages - 1
+                            renderBook()
+                        }, 250)
+                    }
+                }
+                bookPagerView.adapter!!.itemCount - 1 -> {
+                    if (book.currentChapter < book.numChapters - 1) {
+                        bookPagerView.isUserInputEnabled = false
+                        handler.postDelayed({
+                            book.currentChapter = book.currentChapter + 1
+                            book.currentPage = 0
+                            renderBook()
+                        }, 250)
+                    }
+                }
+                else -> {
+                    book.currentPage = if (layoutStateContainer.layoutMode == LayoutMode.NORMAL) {
+                        position - 1
+                    } else {
+                        (2 * (position - 1))
+                    }
+                }
+            }
+        }
+    }
+
+    inner class PageTransformer : ViewPager2.PageTransformer {
+        override fun transformPage(page: View, position: Float) {
+            page.apply {
+                when {
+                    position < -1 -> {
+                        translationX = 0f
+                        translationZ = 0f
+                        alpha = 0f
+                    }
+                    position > 1 -> {
+                        translationX = 0f
+                        translationZ = -1f
+                        alpha = 0f
+                    }
+                    position > 0 -> {
+                        translationX = width * -position
+                        translationZ = -1f
+                        alpha = 1f
+                    }
+                    else -> {
+                        translationX = 0f
+                        translationZ = 0f
+                        alpha = 1f
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.book_menu, menu)
+        val subMenu = menu?.findItem(R.id.action_chapter_group)?.subMenu
+        for (chapter in 0 until book.numChapters) {
+            subMenu?.add("Section $chapter")
+        }
+        subMenu?.add("Accreditation")
```

Comment:
I think "Attribution" would be a better term

Also - all these hardcoded strings should be in `strings.xml` so we can translate (eg "Section")

Reply:
done